src/engagement_db_to_analysis/engagement_db_to_analysis.py

```diff
@@ -156,6 +160,262 @@ def _fold_messages_by_uid(user, messages_traced_data):
+
+    :param analysis_dataset_configs: Dataset configurations to search for the one that relates to the given message.
+    :type analysis_dataset_configs: list of src.engagement_db_to_analysis.configuration.AnalysisDatasetConfiguration
+    :param message: Message to retrieve the analysis dataset configuration for.
+    :type message: engagement_database.data_models.Message
+    :return: Analysis dataset configuration to use for this message.
+    :rtype: src.engagement_db_to_analysis.configuration.AnalysisDatasetConfiguration
+    """
+    for config in analysis_dataset_configs:
+        if message.dataset in config.engagement_db_datasets:
+            return config
+    raise ValueError(f"No analysis dataset configuration found for message '{message.message_id}', which has engagement"
+                     f"db dataset {message.dataset}")
+
+
+def _get_normalised_labels_for_code_scheme(message, code_scheme):
+    """
+    Gets the labels assigned to this message under the given `code_scheme` (or a duplicate of this code scheme),
+    with label scheme_id's normalised to the primary code scheme id in cases where the code scheme was duplicated
+    e.g. scheme_id 'scheme-abc123-1' will be re-written to 'scheme-abc123'.
+
+    :param message: Message to get the labels from.
+    :type message: engagement_database.data_models.Message
+    :param code_scheme: Code scheme to get the latest labels for.
+    :type code_scheme: core_data_modules.data_models.CodeScheme
+    :return: List of the relevant, normalised labels for the given code scheme.
+    :rtype: list of core_data_modules.data_models.Label
+    """
+    relevant_labels = []
+    for label in message.get_latest_labels():
+        if label.scheme_id.startswith(code_scheme.scheme_id):
+            label.scheme_id = code_scheme.scheme_id
+            relevant_labels.append(label)
+    return relevant_labels
+
+
+def _add_message_to_column_td(user, message_td, column_td, analysis_config):
+    """
+    Adds a message to a "column-view" TracedData object in-place.
+
+    This function adds this message's text and labels fields to an existing TracedData object in column-view format.
+    In cases where data already exists in the column-view columns:
+     - Raw texts are handled by concatenation, by `core_data_modules.util.fold_traced_data.FoldStrategies.concatenate`.
+     - Labels are handled by `core_data_modules.util.fold_traced_data.FoldStrategies.list_of_labels`.
+
+    :param user: Identifier of user running the pipeline.
+    :type user: str
+    :param message_td: TracedData representing the message to add
+    :type message_td: core_data_modules.traced_data.TracedData
+    :param column_td: An existing TracedData object in column-view format, to which the relevant data from this message
+                      will be appended.
+    :type column_td: core_data_modules.traced_data.TracedData
+    :param analysis_config: Analysis configuration.
```

Comment:
hmm I think this should be analysis_config because it only accepts one AnalysisConfiguration.

Reply:
I think we accept all here then compute the one that the message belongs to in l211.

Reply:
I think the confusion here and at https://github.com/AfricasVoices/Test-Pipeline-Engagement-DB/pull/70/files#r682720732 is because the function is accepting a single, top-level AnalysisConfiguration but only using the AnalysisConfiguration.DatasetConfigurations. I'll change this to take arg analysis_dataset_configs instead